common.blocks/select/_mode/select_mode_radio.bemhtml

```diff
@@ -1,12 +1,12 @@
 block('select').mod('mode', 'radio')(
-    def().match(this._checkedOptions)(function() {
+    def().match(function() { return this._checkedOptions; })(function() {
         var checkedOptions = this._checkedOptions,
             firstOption = this._firstOption;
         if(firstOption && !checkedOptions.length) {
             firstOption.checked = true;
             checkedOptions = [firstOption];
         }
-        applyNext({ _checkedOption : checkedOptions[0] });
+        return applyNext({ _checkedOption : checkedOptions[0] });
     }),
 
     content()(function() {
```

Comment:
please take a look at https://github.com/tadatuta/select-template-issue-demo
not the very best test case but hope it will help